### ISSUE TITLE
Consolidation of Mapping Change Suggestions

### ISSUE THREAD

**Issue by kenlunde** [adobe-fonts/source-han-serif#37]
This issue is meant for tracking and submitting suggestions for mapping changes, meaning that a character might be better mapped to a different but existing glyph. Note that mapping changes, especially for ideographs, will trigger changes to GSUB features, such as the language-specific lookups of the 'locl' GSUB feature. Because tools are used to build the language-specific lookups of the 'locl' GSUB feature by using the CMap resources, such suggestions cannot be accepted as pull requests, and should instead be posted here. Issues that were submitted before this consolidation issue was opened are referenced by issue number.

* Map U+3164 HANGUL FILLER to _uni1160_.
* Map U+4EBD 亽 to _uni4EBD-CN_ in the JP—and by extension KR—CMap resource per Issue #34.
* Map U+5173 关 to _uni5173-CN_ in the KR CMap resource per Issue #5.
* Map U+5BE7 寧 to _uni5BE7uE0100-JP_ in the KR CMap resource per Issue #20.
* Map U+5553 啓 and U+555F 啟 to _uni5553uE0101-JP_ and _uni555F-JP_, respectively, in the TW CMap resource per Issue #13.
* Map U+58F3 壳 to _uni58F3-JP_ in the TW CMap resource per Issue #26.
* Map U+58FE 壾 and U+591A 多 to _uni58FE-JP_ and _uni591A-JP_, respectively, in the TW CMap resource per Issue #27.
* Map U+6902 椂, U+6903 椃, U+6947 楇, U+7171 煱, and U+9BF1 鯱 to _uni6902-JP_, _uni6903-JP_, _uni6947-JP_, _uni7171-JP_, and _uni9BF1-JP_, respectively, in the TW CMap resource per Issue #32.

**Comment by hfhchan** [comment]
![image](https://cloud.githubusercontent.com/assets/8191296/25070080/5737edac-22c6-11e7-8e88-40cf67bb715e.png)
Though they are out of the scope for the TW subset, the JP glyphs for U+77D2 and U+61DC could be used for the TW locale, similar to Issue #26 and #32.

**Comment by kenlunde** [comment]
@hfhchan: Because this is easy to do, it shall be done.

**Comment by tamcy** [comment]
![u 8019_803b_tw](https://cloud.githubusercontent.com/assets/959433/25342102/80d06ca2-293d-11e7-8ab2-ab142a017604.png)

Incorrect mapping for U+8019 耙 and U+803B 耻 in TW. Both should look the same as the JP/KR glyph.

**Comment by kenlunde** [comment]
@tamcy: Noted. Thanks!

**Comment by tamcy** [comment]
Seems that U+9EFD 黽 is incorrectly serving CN glyph (CID 48026) in JP font with language set to "zh-tw". The language specific version SHSerif-TW uses the correct glyph which is CID 48025. Looks like an issue similar to #43?

**Comment by kenlunde** [comment]
@tamcy: Yes, it is similar to #43, and the solution is to map U+2FCC ⿌ to uni9EFD-JP (CID+48025) in the TW CMap resource.

**Comment by hfhchan** [comment]
![image](https://cloud.githubusercontent.com/assets/8191296/25602996/c3a14f36-2f2a-11e7-9c1b-4b989f728fff.png)

The TW glyph for U+5225 should use that of the JP glyph. It is customary for the bottom left component to be joined upwards in the Code Charts, and the overall shape of 11202 looks too wide for TW/HK customary use.

The TW glyph for U+8382 and U+3B6D could borrow the JP glyph as well, since the bottom left protrusion is considerably closer to conventions.

**Comment by kenlunde** [comment]
While these mapping changes are relatively easy to implement, you missed the deadline for the dot release, given the extraordinarily large number of moving parts that are involved. These will need to wait for a subsequent release.

**Comment by kenlunde** [comment]
@hfhchan: I am willing to map U+3B6D 㭭 and U+5225 別 to uni3B6D-JP and uni5225-JP, respectively, in the TW CMap resource, but not U+8382 莂, mainly due to the inappropriate radical shape, which is far more striking.

**Comment by hfhchan** [comment]
@kenlunde, isn't the JP glyph and CN glyph for U+8382 equally inappropriate for TW? Changing the fallback to JP glyph instead of CN would not introduce more inappropriateness, as far as I can tell...

**Comment by kenlunde** [comment]
@hfhchan: Exactly, which is why I suggest leaving the mapping as-is unless a significant number of people also support the change.

**Comment by tamcy** [comment]
Incorrect lookup data for U+90CE 郎 in JP and KR OTCs. When tagged in TW or CN, the font should substitute CID 41693 with CID 41694, but CID 41717 is incorrectly served as indicated in the jp2cn and jp2tw tables.

![shserif-u90ce](https://cloud.githubusercontent.com/assets/959433/25706309/ac08f182-3112-11e7-84e9-72398a7a5d1c.png)

**Comment by kenlunde** [comment]
@tamcy: This issue is caused an incorrect CN CMap resource override, which inadvertently mapped U+F92C 郎 to _uni90DE-CN_ instead of to _uni90CE-CN_. Keep in mind that substitutions occur at the GID level, so the fact that the mapping for U+F92C 郎 was causing this behavior was not apparent.

**Comment by kenlunde** [comment]
@tamcy: I torpedoed my earlier reply, because the situation is quite complex, and involves KS X 1001 and GB 18030.

U+F92C 郎 (its canonical equivalent is U+90CE 郎) is a GB 18030 character, and its representative glyph is the same as _uni90DE-CN_, meaning with the extra stroke, so its code point maps to that CN glyph. U+F92C 郎 was a KS X 1001 character. Its K source was moved to U+FA2E 郞 with U+90DE 郞 being its canonical equivalent.

Because this will be addressed after the Version 1.001 update, I will give this some more thought. The work-around, especially if you're using the OTCs, is to explicitly select the CN font in the OTC.

**Comment by hfhchan** [comment]
Consider removing jp glyph for U+85F2:

![image](https://cloud.githubusercontent.com/assets/8191296/26076265/c448c7b4-39ea-11e7-8425-354a288e0fe3.png)

**Comment by kenlunde** [comment]
The JP glyph for U+85F2 藲, _uni85F2-JP_, is already a candidate for removal.

**Comment by Jstamz** [comment]
![Glyph001](https://user-images.githubusercontent.com/89553452/132442579-9388cdff-6bf8-44f3-8909-2216cf287a34.png)
I found 12 KR glyphs of version 1.001 which are quite different to those in common Korean fonts.
(함초롬바탕/HCR Batang: The default font of Hangul word processor)
(HY 신명조, HY 견명조: Those fonts are common serif fonts, bundled in MS Office/Hangul Office)
(나눔고딕: Popular free sans serif font)
(한양해서: Widely used Regular script font in Korea)

I suggest to remap 8 of 12 glyphs above: 咎(u+548e), 嗤(u+55e4), 憊(u+618a), 潤(u+6f64), 竇(u+7ac7), 續(u+7e8c), 讀(u+8b80), 隙(u+9699). (Reason: Those glyphs are minor variants in Korea)

(1) I think the mapping for this letter is referring to an incorrect glyph (I didn't see any other fonts using this glyph.)
潤(u+6f64): KR -> JP (*I think the KR glyphs for U+6F64 is both incorrect in Source Han Sans and Source Han Serif)

(2) KR glyphs for these letters are rarely used in Korea; JP glyphs is more often, widely used and similar to the traditional* glyph.
(*These letters are phono-semantic compounds; the common phonetic component is 𧶠, not 賣.)
竇(u+7ac7), 續(u+7e8c), 讀(u+8b80).

(3) These KR glyphs are not rare, but are minor variants.
咎(u+548e): KR -> JP
嗤(u+55e4): JP -> CN
憊(u+618a): KR -> JP
隙(u+9699): KR -> JP
